### PR TITLE
Use latest <li> and <cite> fixes as default

### PR DESCRIPTION
**Pull request by poire-z**
Update default rendering styles of:
  - `<li>` (and css 'display:list-item') from css_d_list_item to css_d_list_item_block
  - `<cite>` from css_d_block to css_d_inline

As changing display: properties may change xpointers/xpaths and may break past bookmarks and highlights, increment DOM_VERSION_CURRENT to 20180524, and allow KOReader to request the previous behaviour on previously opened books.
See https://github.com/koreader/koreader/issues/3940#issuecomment-391667791

Also remove the handling of PROP_DOM_VERSION as a property as this was a bad idea (the last one was saved in cr3.ini, and we would always get that one back). Base will now use specific methods to get gDOMVersionCurrent and update gDOMVersionRequested.

### CODE REVIEW

File: cr3gui/data/fb2.css [hunk 1]
```diff
@@ -25,7 +25,7 @@ stanza { margin-left: 15%; text-align: left; font-style: italic  }
 poem { margin-top: 1em; margin-bottom: 1em; text-indent: 0px }
 text-author { font-weight: bold; font-style: italic; margin-left: 5%}
 epigraph { margin-left: 15%; margin-right: 1em; text-align: left; text-indent: 1em; font-style: italic; margin-top: 15px; margin-bottom: 25px }
-cite { font-style: italic; margin-left: 5%; margin-right: 5%; text-align: justify; margin-top: 20px; margin-bottom: 20px }
+cite { display: block; font-style: italic; margin-left: 5%; margin-right: 5%; text-align: justify; margin-top: 20px; margin-bottom: 20px }
```
Comment (Frenzie) [inline cr3gui/data/fb2.css]: You didn't mention this in the commit message. This is a Good Idea™? :-)

Comment (poire-z) [reply cr3gui/data/fb2.css]: I'll add it when Squash and merge: `Keep CITE as a block element for FB2 (where it is defined as being a container) only via fb2.css`.
Why the `?`, you have doubts this is a good idea?
The thing is that it's easier for me to implement previous behaviour when we change things in fb2def.h. If we change things via css, I can't detect the change.
With this fb2.css change, old and new fb2 files will have CITE being display: block, so it should be fine.

Comment (Frenzie) [reply cr3gui/data/fb2.css]: No, I'm just saying i have no idea about FB2 so I don't really know what cite should look like there or what it does. So between the lines, whether you've verified all that.

Comment (poire-z) [reply cr3gui/data/fb2.css]: I haven't verified it on a real fb2.
But it looks like all our .css files (including some css hardcoded in src/lvdocview.cpp) have as an ancestor the fb2.css, with tag names unknown to HTML, but defined in the fb2 specs that @cramoisi found https://github.com/koreader/crengine/pull/191#issuecomment-391662414.
And this schema seems to indicate CITE is a container, containing elements P, POEM, EMPTY-LINE.
So, we know where the tag names come from, and why CITE was a block element.
So, let's just keep it a block element for FB2 only.

**Comment by poire-z** [comment]
I have this that solves the `<tagada>` situation described in https://github.com/koreader/crengine/pull/191#issuecomment-391742834 , with no side effect seen for now:

```c
diff --git a/crengine/include/lvstyles.h b/crengine/include/lvstyles.h
index 7df78fac..a8b02c31 100644
--- a/crengine/include/lvstyles.h
+++ b/crengine/include/lvstyles.h
@@ -131,7 +131,7 @@ typedef struct css_style_rec_tag {
     : refCount(0)
     , hash(0)
     , important(0)
-    , display( css_d_inherit )
+    , display( css_d_inline )
     , white_space(css_ws_inherit)
     , text_align(css_ta_inherit)
     , text_align_last(css_ta_inherit)
diff --git a/crengine/src/lvrend.cpp b/crengine/src/lvrend.cpp
index a760d8be..1f88cfb4 100755
--- a/crengine/src/lvrend.cpp
+++ b/crengine/src/lvrend.cpp
@@ -2855,6 +2855,13 @@ void setNodeStyle( ldomNode * enode, css_style_ref_t parent_style, LVFontRef par
     css_style_ref_t style( new css_style_rec_t );
     css_style_rec_t * pstyle = style.get();

+    if (gDOMVersionRequested < 20180524) {
+        // The display property initial value has been changed from css_d_inherit
+        // to css_d_inline (as per spec, and so that an unknown element does not
+        // become block when contained in a P, and inline when contained in a SPAN)
+        pstyle->display = css_d_inherit;
+    }
+
 //    if ( parent_style.isNull() ) {
 //        CRLog::error("parent style is null!!!");
 //    }
@@ -2938,7 +2945,9 @@ void setNodeStyle( ldomNode * enode, css_style_ref_t parent_style, LVFontRef par
         //parent_style->text_align = css_ta_center;
     //}

-    UPDATE_STYLE_FIELD( display, css_d_inherit );
+    if (gDOMVersionRequested < 20180524) { // display should not be inherited
+        UPDATE_STYLE_FIELD( display, css_d_inherit );
+    }
     UPDATE_STYLE_FIELD( white_space, css_ws_inherit );
     UPDATE_STYLE_FIELD( text_align, css_ta_inherit );
     UPDATE_STYLE_FIELD( text_decoration, css_td_inherit );
```

Dunno if I should include that in this PR (to not multiply the dom_version changes), or if I should do it in another PR for easier bissecting in case of strange things.

**Comment by Frenzie** [comment]
Might as well tack it on I suppose? It seems like a simple, straightforward fix.

**Comment by poire-z** [comment]
There's something with the last commit from this PR...
I noticed some kind of degradation of table rendering (degradation as "no table rendering at all" :) on some Wikipedia page, since I changed unknown elements `display:` style from `inherit` to `inline`.

It turns out that, since before my recent changes, crengine seems to not render tables when they are in some inline element.
Sample HTML:
```html
<div>
<table style="border: black 1px solid">
<tr><td>cell 11</td><td>cell 12</td></tr>
<tr><td>cell 21</td><td>cell 22</td></tr>
</table>
</div>
```
Nice output:
<kbd>image</kbd>

Add some inline element in between:
`<div><span> <table stuff> </span></div>`
Ugly output:
<kbd>image</kbd>

So, I guess it's an historical limitation of crengine.

It's now a problem with the quite common: `<div><center> <table stuff> </center></div>` , CENTER being an unknown element.

Before, if you added an unknow element, it inherited its `display:` from the DIV (or BODY) , so it was block, and we had _Nice output_.
Now, unknown elements don't inherit and are inline. So _Ugly output_.

Haven't yet looked at if this table in inline problem can be fixed (I never witnessed this problem, so, if I'm not wrong about it being historical, it's that it's very rare to put a table in some inline element, and people have the natural habbit to close the inline stuff before a table).
This displays as _nice output_:
`<div> some text <span> more text </span> <table stuff> </div>`
Not this:
`<div> some text <span> more text  <table stuff> </span> </div>`


But should I add CENTER to #196 block element additions for a quick fix of this common case?
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/center
Other ideas of obsolete elements similar to CENTER that we should still treat as block elements?

**Comment by cramoisi** [comment]
Arg, even when it wasn't  deprecated, I didn't like `<center>` But even so, we're bound to find it here or there.
`menu` ?
`dir` ?

**Comment by poire-z** [comment]
Adding CENTER, MENU and DIR should't hurt.
XMP could be like PRE (with may be some differences in escaping, so I'm not sure)

I'm not sure APPLET is a block element, I guess it's like OBJECT which can be inline.
PLAINTEXT is not implemented by most browsers (and it stops HTML processing), so this one is out.

About crengine rendering inline children as inline, the code does that explictely.
It sets render methods deep first (children first), and then when processing upper elements, there's this:
https://github.com/koreader/crengine/blob/3142ef34ede3d9719accd2f5e8e05049d1b4fb44/crengine/src/lvtinydom.cpp#L4150-L4153
Just not doing that messes with many other renderings, so ... no simple fix.
Just some limitation to remember.

**Comment by Frenzie** [comment]
> I'm not sure APPLET is a block element, I guess it's like OBJECT which can be inline.

Neither am I but I thought you mentioned object? It was just a quick otoh note of things to investigate. :)

**Comment by poire-z** [comment]
No prob :) #196 updated with CENTER, MENU, DIR and XMP.

**Pull request merged**